### PR TITLE
feat(south-sqldbtofile): add sqlite driver for local sqlite db file

### PR DESCRIPTION
**Pull request by burgerni10**
This is a Work In Progress PR : the front still need to be improved to display only the appropriate settings depending on the selected driver.

For example : 
-sqlite driver only has databasePath and no host/port etc
-mssql has host, port, domain...
-etc

### CODE REVIEW

**Comment by burgerni10** [comment]
See issue #1160

**Comment by jfhenon** [comment]
@burgerni10 , your PR seems OK to me but I would prefer @kukukk to review your code as he was the initial developer on this South.

File: src/south/SQLDbToFile/SQLDbToFile.class.js [hunk 1]
```diff
@@ -228,97 +233,125 @@ class SQLDbToFile extends ProtocolHandler {
...
+  async getDataFromSqlite() {
+    const adaptedQuery = this.query.replace('@LastCompletedDate', '?')
+    this.logger.debug(`Executing "${adaptedQuery}" ${this.containsLastCompletedDate ? 'with' : 'without'} LastCompletedDate`)
+
+    let database = null
+    let data = []
+    try {
+      database = await sqlite.open({ filename: this.databasePath, driver: sqlite3.Database })
+      const stmt = await database.prepare(adaptedQuery)
+      if (this.containsLastCompletedDate) {
```
Comment (kukukk) [inline src/south/SQLDbToFile/SQLDbToFile.class.js:305]: I like one-liners :}
`const preparedParameters = this.containsLastCompletedDate ? { '@LastCompletedDate': new Date(this.lastCompletedAt) } : {}`
`data = await stmt.all(preparedParameters)`

**Review by kukukk — COMMENTED** [review]
I see that you changed the order of the methods (moved getDataFromMySQL to the end), but git seems to be not smart enough to detect it, and shows a lot of changes in the file, while it should only have some additions for SQLite.

**Comment by burgerni10** [comment]
> I see that you changed the order of the methods (moved getDataFromMySQL to the end), but git seems to be not smart enough to detect it, and shows a lot of changes in the file, while it should only have some additions for SQLite.

I moved it while improving testing and I thought GitHub would detect it. Now that I know that, I'll avoid that in future commits :)

File: src/south/SQLDbToFile/SQLDbToFile.class.js [hunk 2]
```diff
@@ -228,97 +233,125 @@ class SQLDbToFile extends ProtocolHandler {
...
-      )
+      const { rows } = await connexion.query(adaptedQuery, params)
       data = rows
     } catch (error) {
       this.logger.error(error)
     } finally {
-      if (connection) {
-        await connection.end()
-      }
+      await connexion.end()
```
[the remark this answers is not in the document]
Comment (burgerni10) [reply src/south/SQLDbToFile/SQLDbToFile.class.js:251]: You're right, I removed it while improving testing and forgot to put it back. Thanks !

**Review by kukukk — APPROVED** [review]

**Pull request merged**